File: Backend/Livro.cs
namespace WebApplication1
{
    public class Livro
    {
        public string Titulo { get; set; }
        public string Autor { get; set; }
        public DateOnly DataPublicacao { get; set; }
        public List<Genero> Generos { get; set; }
        public int Copias { get; set; }
        

        public Livro(string titulo)
        {
            Titulo = titulo;
        }
    }
}

File: Frontend/src/app/models/livro.ts
import {Genero} from "./genero";

export class Livro {
  id: number = 0;
  titulo: string = "";
  autor: string = "";
  dataPublicacao: Date = new Date();
  copias: number = 0;
  generos: Genero[] = [];

  constructor(titulo: string, autor: string, dataPublicacao: Date,
              generos: Genero[], copias: number = 0) {
    this.titulo = titulo;
    this.autor = autor;
    this.dataPublicacao = dataPublicacao;
    this.copias = copias;
    this.generos = generos;
  }
}
